Crosscompilation for Windows under Ubuntu fails

### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Ubuntu 18.04.2 LTS (WSL)
| File format of the file you reverse (mandatory)      | n/a
| Architecture/bits of the file (mandatory)            | n/a
| r2 -v full output, **not truncated** (mandatory)     | radare2 4.0.0 23255 @ linux-x86-64 git.4.0.0 commit: e97047299af43e9b4e658c246816a2b48bdc087e build: 2019-11-12__21:50:29

### Expected behavior
Cross-compilation succeeds.

### Actual behavior
Cross-compilation fails with an error.

### Steps to reproduce the behavior 
- follow the steps for cross compilation from https://github.com/radareorg/radare2/blob/master/doc/windows.md for Ubuntu

### Additional Logs, screenshots, source-code,  configuration dump, ...
Error during make:
> i586-mingw32msvc-gcc -c  -DUNICODE -D_UNICODE  -MD -D__WINDOWS__=1    -g -Wall -D__WINDOWS__=1 -DR2_PLUGIN_INCORE -I/home/ubuntu/radare2/libr/..//shlr -I/home/ubuntu/radare2/libr/../shlr/zip/include -I/home/ubuntu/radare2/libr -I/home/ubuntu/radare2/libr/include -fvisibility=hidden -I../../shlr/sdb/src//src -DHAVE_FORK=1 -o mem.o mem.c
In file included from /home/ubuntu/radare2/libr/include/r_util.h:13,
                 from mem.c:3:
/home/ubuntu/radare2/libr/include/r_th.h:58: error: expected specifier-qualifier-list before ‘CONDITION_VARIABLE’
mem.c: In function ‘r_mem_memzero’:
mem.c:335: warning: implicit declaration of function ‘explicit_bzero’

--- a/doc/windows.md
+++ b/doc/windows.md
@@ -1,56 +1,34 @@
-HOW TO BUILD FOR WINDOWS
+How to Build for Windows
 ========================
-Actually, if you want to build r2 for windows, you can use native compiler with mingw32 or
-cygwin, but there's also the possibility to crosscompile from a Linux box:
 
+You can follow the [r2book](https://radare.gitbooks.io/radare2book/content/first_steps/windows_compilation.html) for a more complete guide.
+
+Native
+---------------
+
+You will need:
+
+* Python 3
+* Meson (pip3 install meson)
+* Visual Studio 2015 (or later)
+
+	
+First, call `vcvarsall.bat` with your architecture (x86, x64, arm) to setup the compilation environment.
+
+	cd radare2	
+	python3 sys\meson.py --release --backend vs2019 --shared --install="%cd%\radare2_dist" --webui
+
+You can change `--backend` to your VS version (`vs2015`, `vs2017`), `ninja` buildsystem is also supported.
+
+For XP support, append `--xp` to the command (not compatible with VS2019).
+
+You can then add `radare2_dist` to your PATH to make radare2 accessible from everywhere.
 
 Crosscompilation
 ----------------
 
-**ArchLinux**
+As building with mingw is no longer officially supported in radare2, crosscompilation isn't (easily) possible.
 
-	./configure --with-compiler=i486-mingw32-gcc --with-ostype=windows --host=i486-unknown-windows
-	make
-	make w32dist
+You can check the official [Meson documentation](https://mesonbuild.com/Cross-compilation.html).
 
-	cd swig
-	export CC=i486-mingw32-gcc
-	export CXX=i486-mingw32-g++
-	unset CFLAGS
-	unset LDFLAGS
-	export LDFLAGS="-L${HOME}/.wine/drive_c/Python27/libs ${LDFLAGS}"
-	export CFLAGS="-Wl,--enable-auto-import -L../../radare2-w32-0.5 ${CFLAGS}"
-	./configure --with-ostype=windows --host=i486-unknown-windows --with-cc=i486-mingw32-gcc --with-cxx=i486-mingw32-g++
-
-  
-**Ubuntu**
- 
-	sudo apt-get install mingw32
-	./configure --with-compiler=i586-mingw32msvc-gcc  --with-ostype=windows --host=i586-unknown-windows
-	make
-	make w32dist
-
-**Windows x86-64**
- 
-***Setup toolchain***
-    
-	open http://mingw-w64.sourceforge.net/
-	wget http://switch.dl.sourceforge.net/project/mingw-w64/Toolchains%20targetting%20Win64/Automated%20Builds/mingw-w64-bin_i686-linux_20110627.tar.bz2
-	mkdir -p mingw64
-	tar xjvf mingw-w64*.bz2 -C mingw64
-	export PATH=$PWD/mingw64/bin:$PATH
-  
-Build radare2
--------------
-  
-	./configure --with-compiler=x86_64-w64-mingw32-gcc --with-ostype=windows --host=x86_64-unknown-windows
-
-
-Python bindings:
-----------------
-
-	wget http://www.python.org/ftp/python/2.7/python-2.7.msi
-	msiexec /i python-2.7.msi
-	cd r2-bindings
-	make w32
-	make w32dist
+Good luck.

